tests/cnf/core/network/sriov/tests/metricsExporter.go

```diff
@@ -0,0 +1,472 @@
+			Expect(sriovDeviceID).ToNot(BeEmpty(), "Expected sriovDeviceID not to be empty")
+
+			By("Enable Sriov Metrics Exporter feature in default SriovOperatorConfig CR")
+			setMetricsExporter(true)
+
+			By("Verify new daemonset sriov-network-metrics-exporter is created and ready")
+			Eventually(func() bool {
+				sriovmetricsdaemonset, err = daemonset.Pull(
+					APIClient, "sriov-network-metrics-exporter", NetConfig.SriovOperatorNamespace)
+
+				return err == nil
+			}, 2*time.Minute, 2*time.Second).Should(BeTrue(), "Daemonset sriov-network-metrics-exporter is not created")
+			Expect(sriovmetricsdaemonset.IsReady(2*time.Minute)).Should(BeTrue(),
+				"Daemonset sriov-network-metrics-exporter is not ready")
+
+			By("Enable Prometheus scraping for the new Sriov Metrics Exporter by labeling operator namespace")
+			sriovNs, err := namespace.Pull(APIClient, NetConfig.SriovOperatorNamespace)
+			Expect(err).ToNot(HaveOccurred(), "Failed to fetch Sriov Namespace")
+			_, err = sriovNs.WithMultipleLabels(netparam.ClusterMonitoringNSLabel).Update()
+			Expect(err).ToNot(HaveOccurred(), "Failed to update Sriov Namespace")
+		})
+
+		AfterEach(func() {
+			By("Removing SR-IOV configuration")
+			err := sriovenv.RemoveSriovConfigurationAndWaitForSriovAndMCPStable()
+			Expect(err).ToNot(HaveOccurred(), "Failed to remove SR-IOV configration")
+
+			By("Cleaning test namespace")
+			err = namespace.NewBuilder(APIClient, tsparams.TestNamespaceName).CleanObjects(
+				netparam.DefaultTimeout, pod.GetGVR())
+			Expect(err).ToNot(HaveOccurred(), "Failed to clean test namespace")
+		})
+
+		AfterAll(func() {
+			By("Disable Sriov Metrics Exporter feature in default SriovOperatorConfig CR")
+			setMetricsExporter(false)
+			Eventually(func() bool { return sriovmetricsdaemonset.Exists() }, 1*time.Minute, 1*time.Second).Should(BeFalse(),
+				"sriov-metrics-exporter is not deleted yet")
+
+		})
+
+		Context("Sriov Metrics Exporter - Netdevice to Netdevice", func() {
+			It("Sriov Metrics Exporter - Netdevice to Netdevice - Same PF", reportxml.ID("74762"), func() {
+				runNettoNetTests(sriovInterfacesUnderTest[0], sriovInterfacesUnderTest[0],
+					workerNodeList[0].Object.Name, workerNodeList[0].Object.Name, sriovDeviceID)
+			})
+			It("Sriov Metrics Exporter - Netdevice to Netdevice - Different PF", reportxml.ID("75929"), func() {
+				runNettoNetTests(sriovInterfacesUnderTest[0], sriovInterfacesUnderTest[1],
+					workerNodeList[0].Object.Name, workerNodeList[0].Object.Name, sriovDeviceID)
+			})
+
+			It("Sriov Metrics Exporter - Netdevice to Netdevice - Different Worker", reportxml.ID("75930"), func() {
+				runNettoNetTests(sriovInterfacesUnderTest[0], sriovInterfacesUnderTest[0],
+					workerNodeList[0].Object.Name, workerNodeList[1].Object.Name, sriovDeviceID)
+			})
+		})
+
+		Context("Sriov Metrics Exporter - Netdevice to Vfiopci", func() {
+			BeforeAll(func() {
+				By("Deploying PerformanceProfile is it's not installed")
```

Comment:
typo:
Deploying PerformanceProfile if it's not installed

Reply:
done